grim_test.go

```diff
@@ -8,11 +8,50 @@ import (
 	"os"
 	"path/filepath"
 	"testing"
+	"bytes"
+	"strings"
 )
 
 // Copyright 2015 MediaMath <http://www.mediamath.com>.  All rights reserved.
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
+
+func TestTruncatedGrimServerID(t *testing.T) {
+	var buf bytes.Buffer
+	logger := log.New(&buf, "", log.Lshortfile)
+
+	tempDir, _ := ioutil.TempDir("", "TestTruncatedGrimServerID")
+	defer os.RemoveAll(tempDir)
+
+	empty := "empty"
+	hugeString := "12345678901234567890"
+
+	grimConfig := &config{
+		ResultRoot:    &empty,
+		WorkspaceRoot: &empty,
+		AWSRegion:     &empty,
+		AWSKey:        &empty,
+		AWSSecret:     &empty,
+		GrimServerID:  &hugeString,
+	}
+
+	configJs, _ := json.Marshal(grimConfig)
+	ioutil.WriteFile(filepath.Join(tempDir, "config.json"), configJs, 0644)
+
+	g := &Instance{
+		configRoot : &tempDir,
+		queue      : nil,
+	}
+
+	g.PrepareGrimQueue(logger)
+	message := fmt.Sprintf("%v", &buf)
+
+	if !strings.Contains(message, "GrimServerID shouldn't be over 15 characters and has been truncated") {
```

Comment:
I suggest making this some type of variable that's shared between tests and the actual ouput of the error message in grim so every time someone changes the error message, you don't have to change the test to match the output.

Also above,

`logger.Printf("GrimServerID shouldn't be over 15 characters and has been truncated\nPlease update your config.json file to have a shorter grimServerID\nOr remove the GrimServerID entry to use server defaults\n")`

Could have the same error message, or this one below could have that one.  It keeps everything nice and tidy.